### PR TITLE
Fixes #27687 - Add vhost_params and proxy_pass_params

### PR DESCRIPTION
**Pull request by ekohl**
This allows customizing the Apache vhost and ProxyPass via hiera giving more flexibility. Also does some more cleaning up as preparation.

### CODE REVIEW

File: manifests/reverse_proxy.pp [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
+# @param servername
+#   The Apache vhost server name to listen on
+# @param port
+#   The port to listen on
+# @param ssl_protocol
+#   The ssl protocol(s) to accept
+# @param vhost_params
+#   Any parameters to pass to the apache::vhost resource
+# @param proxy_pass_params
+#   Any parameters to pass to the apache::vhost resource
```
Comment (evgeni) [inline manifests/reverse_proxy.pp]:
```suggestion
#   Any parameters to pass to the proxy_pass param of the apache::vhost resource
```

File: manifests/reverse_proxy.pp [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
 class foreman_proxy_content::reverse_proxy (
-  $path = '/',
-  $url  = "${foreman_proxy_content::foreman_url}/",
-  $port = $foreman_proxy_content::reverse_proxy_port,
-  $ssl_protocol = undef,
+  String $path = '/',
```
Comment (evgeni) [inline manifests/reverse_proxy.pp]: Is there a better type to denote "an absolute (looking) path"?

Comment (ekohl) [reply manifests/reverse_proxy.pp]: There is `Stdlib::Unixpath` which is strictly speaking not a URL path, but might be good enough.

Comment (evgeni) [reply manifests/reverse_proxy.pp]: Yeah, I thought about Stdlib::Absolutepath, but yours is better.

**Comment by ekohl** [comment]
Updated

**Pull request merged**